### PR TITLE
Oll 36 setup user connection

### PR DESCRIPTION
**Pull request by Toumi-Elyes**
## Summary

Routes to update and delete users and middleware implementation with jwt.

## Checklist

**Bureaucracy**

- [x] I set a descriptive title
- [x] I set a proper description, providing enough information to the reviewer
- [x] I considered squashing this PR, if needed
- [ ] I set the proper assignee
- [ ] I set the proper reviewers

**Quality Assurance**

- [x] I tested the behavior of this PR locally and made sure it complies with
  the ticket description

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -7,6 +7,11 @@ datasource db {
...
 }
 
+enum Role {
+  MEMBER
+  ADMIN
+}
+
 model User {
   id             String         @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
   firstname      String
```
Comment (ThisisYoYoDev) [inline backend/prisma/schema.prisma:17]: can be replace by @uuid
`
id String @id @default(uuid()) @db.Uuid
`

File: backend/src/user/user.controller.ts [hunk 2]
```diff
@@ -26,9 +35,78 @@ export class UserController {
...
+          Email: 'test@test.test',
+          Password: '1234',
+        } as LoginUserModel,
+      },
+    },
+  })
+  @Post('/login')
+  async loginUser(@Body() body: LoginUserModel): Promise<string> {
+    const token = this.userService.loginUser(body);
+    return token;
```
Comment (ThisisYoYoDev) [inline backend/src/user/user.controller.ts]: return direcly the function

File: backend/src/user/user.dto.ts [hunk 3]
```diff
@@ -7,11 +7,9 @@ export class UserModel {
   Id: string;
 
   @ApiProperty()
-  @IsString()
```
Comment (ThisisYoYoDev) [inline backend/src/user/user.dto.ts]: why you remove IsString()

File: backend/src/user/user.dto.ts [hunk 4]
```diff
@@ -25,8 +23,26 @@ export class UserModel {
   Communities_id: string[];
 }
 
-export class RegisterUserModel {
+export class JwtUserModel {
+  @IsUUID()
+  Id: string;
+
+  @ApiProperty()
```
Comment (ThisisYoYoDev) [inline backend/src/user/user.dto.ts]: Do you use it in the swagger ?
if no remove ApiProperty

File: backend/src/user/user.dto.ts [hunk 5]
```diff
@@ -25,8 +23,26 @@ export class UserModel {
...
+  Firstname: string;
```
Comment (ThisisYoYoDev) [inline backend/src/user/user.dto.ts:33]: Should be IsString()

File: backend/src/user/user.service.ts [hunk 6]
```diff
@@ -47,6 +54,13 @@ export class UserService {
   }
 
   async registerUser(userData: RegisterUserModel): Promise<string> {
+    if (userData.Firstname === '' || userData.Lastname === '') {
```
Comment (ThisisYoYoDev) [inline backend/src/user/user.service.ts]: add check for undefined

**Review by ThisisYoYoDev — CHANGES_REQUESTED** [review]

File: backend/prisma/schema.prisma (hunk 1 above)
Comment (Toumi-Elyes) [reply backend/prisma/schema.prisma:17]: fixed

File: backend/src/middleware.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import * as jwt from 'jsonwebtoken';
+import prisma from 'client';
+
+@Injectable()
+export class LoggingMiddleware implements NestMiddleware {
+  use(req: Request, _: Response, next: NextFunction) {
+    const token: string = req.headers.authorization_token as string;
+    console.log('token = ', token);
+
+    if (token === undefined) {
```
Comment (ThisisYoYoDev) [inline backend/src/middleware.ts]: better syntax with if (!token)

File: backend/src/user/user.service.ts [hunk 8]
```diff
@@ -84,4 +102,84 @@ export class UserService {
...
+        id: parsedJwt['id'],
+      },
+    });
+
+    if (!userDb) {
+      Logger.error('User does not exists !');
+      throw new NotFoundException('User does not exists !');
+    }
+
+    return 'Ok.';
```
Comment (ThisisYoYoDev) [inline backend/src/user/user.service.ts]: replace by real message

**Review by ThisisYoYoDev — CHANGES_REQUESTED** [review]

**Review by ThisisYoYoDev — APPROVED** [review]
LGTM, when the CI pass

**Pull request merged**